templates/pwm.compound.json

```diff
@@ -0,0 +1,308 @@
+			"MinLength": "1",
+			"Type": "String"
+		},
+		"MaxCapacity": {
+			"Default": "1",
+			"Description": "The maximum number of PWM instances for the autoscale group",
+			"MinLength": "1",
+			"Type": "String"
+		},
+		"PrivateSubnetIDs": {
+			"Description": "List of Private Subnet IDs where the PWM instances will run. (only select one, load balancing not tested)",
+			"Type": "List<AWS::EC2::Subnet::Id>"
+		},
+		"PublicSubnetIDs": {
+			"Description": "A list of Public subnet IDs to attach to the load balancer.",
+			"Type": "List<AWS::EC2::Subnet::Id>"
+		},
+		"ConfigBucketName": {
+			"Description": "Name of the existing S3 bucket where the PWM config should be pulled and pushed",
+			"Type": "String"
+		},
+		"GitHubUsername": {
+			"Description": "The username for the GitHub repo that will be cloned",
+			"Type": "String"
+		},
+		"GitHubRepo": {
+			"Description": "The GitHub repo that will be cloned",
+			"Type": "String"
+		},
+		"SaltPillarFile": {
+			"Description": "Name of the pillar file that's stored in the S3 config bucket",
+			"Type": "String"
+		},
+		"SNSSubscriptionEmail": {
+			"AllowedPattern": "^([a-zA-Z0-9_\\-\\.]+)@([a-zA-Z0-9_\\-\\.]+)\\.([a-zA-Z]{2,5})$",
+			"Default": "deployer@example.com",
+			"Description": "Email address to which AWS SNS events are sent.",
+			"MinLength": "3",
+			"Type": "String"
+		},
+		"VPC": {
+			"Description": "VPC ID",
+			"Type": "AWS::EC2::VPC::Id"
+		}
+	},
+	"Resources": {
+		"PWMAutoScaleStack": {
+			"Properties": {
+				"Parameters": {
+					"AdminIAMGroup": {
+						"Ref": "AdminIAMGroup"
+					},
+					"AmiId": {
+						"Ref": "AmiId"
+					},
+					"DesiredCapacity": {
+						"Ref": "DesiredCapacity"
+					},
+					"DiscoIAMGroup": {
+						"Ref": "DiscoIAMGroup"
+					},
+					"InstanceType": {
+						"Ref": "InstanceType"
+					},
+					"KeyPairName": {
+						"Ref": "KeyPairName"
+					},
+					"MaxCapacity": {
+						"Ref": "MaxCapacity"
+					},
+					"MinCapacity": {
+						"Ref": "MinCapacity"
+					},
+					"PrivateSubnetIDs": {
+						"Fn::Join": [
+							",",
+							{
+								"Ref": "PrivateSubnetIDs"
+							}
+						]
+					},
+					"PWMALBTargetGroupName": {
+						"Fn::GetAtt": [
+							"PWMALBStack",
+							"Outputs.TargetGroupName"
+						]
+					},
+					"PWMSNSTopicARN": {
+						"Fn::GetAtt": [
+							"PWMSNSTopicSubscriptStack",
+							"Outputs.SNSTopicARN"
+						]
+					},
+					"ConfigBucketName": {
+						"Ref": "ConfigBucketName"
+					},
+					"GitHubUsername": {
+						"Ref": "GitHubUsername"
+					},
+					"GitHubRepo": {
+						"Ref": "GitHubRepo"
+					},
+					"SaltPillarFile": {
+						"Ref": "SaltPillarFile"
+					},
+					"SecurityGroupIdPWMInstance": {
+						"Fn::Join": [
+							",",
+							[{
+								"Fn::GetAtt": [
+									"PWMALBStack",
+									"Outputs.PWMSecurityGroupId"
+								]
+							}]
+						]
+					}
+				},
+				"TemplateURL": "https://s3.amazonaws.com/gowens-pwm/pwm.asg.element.json"
```

Comment:
These template urls will need to change before merging...

Reply:
_Highly_ recommend looking at `aws cloudformation package` or `aws cloudformation deploy` to avoid hard-coding the s3 paths here.